integration/setup/startup.go

```diff
@@ -53,7 +53,16 @@ func Startup() {
 	prometheus.DefaultRegisterer.MustRegister(listenerMetrics)
 
 	// use any available port to allow running different configurations in parallel
-	go debug.RunHandler(context.Background(), "127.0.0.1:0", prometheus.DefaultRegisterer, zap.L().Named("debug"))
+	h, err := debug.Listen(&debug.ListenOpts{
+		TCPAddr: "127.0.0.1:0",
+		L:       zap.L().Named("debug"),
+		R:       prometheus.DefaultRegisterer,
+	})
+	if err != nil {
+		zap.S().Fatalf("Failed to create debug handler: %s.", err)
+	}
+
+	go h.Serve(context.Background())
```

Comment:
So, we don't shut down the handler in tests? Is it because there is nothing to shutdown, or is it because there is no mechanism to do it?

Reply:
It is mainly to make PR smaller. There is not much to properly shutdown, but since you are going to do it for the Otel exporter (which we must shutdown gracefully), it could be shutdown as well